contrib/hived-config-patcher/src/patcher.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import sys
+import yaml
+import logging
+import argparse
+from kubernetes import client, config, watch
+
+
+class HiveDConfigPatcher(object):
+    def __init__(self, min_nodes=1, max_nodes=2**31-1, hived_config_file=None):
+        logging.basicConfig(level=logging.INFO, format="%(asctime)s [%(levelname)s] %(message)s")
+        self.min_nodes = min_nodes
+        self.max_nodes = max_nodes
+        self.hived_config_file = hived_config_file
+        self.__init_hived_config()
+        self.__init_kube_client()
+        self.__nodes = []
+        self.__fake_nodes = ["fake{}".format(self.base36(i)) for i in range(max_nodes)]
```

Comment:
Why not just i instead of base36(i), for easy read and parse

Reply:
to keep consistent with vmss naming format, [six digits base 36](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-instance-ids#scale-set-vm-computer-name)

Reply:
ok, but seems no benifts to keep this consistent